osdep/macosx_application.m

```diff
@@ -353,6 +353,30 @@ static bool bundle_started_from_finder(int argc, char **argv)
     }
 }
 
+static void init_path_for_bundle()
+{
+    NSTask *shell_task = [[[NSTask alloc] init] autorelease];
+    NSPipe *shell_out = [NSPipe pipe];
+    NSDictionary *env = [[NSProcessInfo processInfo] environment];
+    NSString *shell_path = [env objectForKey:@"SHELL"];
+
+    [shell_task setLaunchPath:shell_path];
+    [shell_task setArguments:@[@"-l", @"-c", @"echo $PATH"]];
+    [shell_task setStandardOutput:shell_out];
+    [shell_task launch];
+    [shell_task waitUntilExit];
```

Comment:
to get the $PATH used by the shell, eg the one mpv uses when started from CLI. i could also just hardcode one instead of trying to retrieve the one configured by the user.
i couldn't find a better way to retrieve that path and yes it's quite WTF.